pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -70,9 +84,50 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, pageInfo)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if rational options are provided.
+// Always return nil error, absorbs logical error provided in query by returning full list of members.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo paginationInfo) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages = 0
+		memberList.CurrentPage = 0
+		return nil
+	}
+	if pageInfo.pageLimit > 0 && pageInfo.pageNumber > 0 {
+		pageInfo.totalPages = int(math.Ceil(math.Max(float64(len(memberList.EffectiveMembers))/float64(pageInfo.pageLimit), float64(pageInfo.totalPages))))
+		if pageInfo.totalPages >= pageInfo.pageNumber {
+			// Sort members based on name of ee/pod to realize consistent pagination support.
+			sort.SliceStable(memberList.EffectiveMembers, func(i, j int) bool {
+				if memberList.EffectiveMembers[i].ExternalEntity != nil && memberList.EffectiveMembers[j].ExternalEntity != nil {
+					return memberList.EffectiveMembers[i].ExternalEntity.Name < memberList.EffectiveMembers[j].ExternalEntity.Name
+				} else if memberList.EffectiveMembers[i].Pod != nil && memberList.EffectiveMembers[j].Pod != nil {
```

Comment:
nit: maybe evaluate `Pod` first as in most cases `Pod` will be the one set

Reply:
👌 makes sense